spec/exmples/user_spec.rb

```diff
@@ -0,0 +1,74 @@
+require File.expand_path(File.dirname(__FILE__) + '../../spec_helper')
+
+describe GoogleContactsApi::User do
+    let(:oauth) { double ("oauth") }
+    let(:api) { double("api") }
+    let(:user) { GoogleContactsApi::User.new(@oauth) }
+
+    # Should hit the right URLs and return the right stuff
+    describe ".groups" do
+      it "should be able to get groups including system groups" do
+        expect(user).to receive(:get_groups).and_return("group set")
+        expect(user.groups).to eq("group set")
+      end
+    end
+    describe ".groups!" do
+      it "should be able to get groups" do
+        expect(user).to receive(:get_groups).and_return("group set")
+        expect(user.groups!).to eq("group set")
```

Comment:
He's made groups! and contacts! to return nil if nothing is there, the normal methods have no such safety net